### PR TITLE
Feature/161 support json rpc

### LINKED ISSUE #161: Allow using json-rpc service
In order to test creation of new buckets (https://github.com/exasol/bucketfs-java/issues/17) we need a way to access the json-rpc service:

1. Forward the default json-rpc port `443` by default, i.e. add it to the exposed ports.
    * This is required for accessing the service.
2. Add a method for getting the default json-rpc port similar as `getDefaultInternalBucketfsPort()`
    * This is a convenience method for getting the default port without hard-coding it.
3. Extend `ClusterConfiguration` to allow reading `Global/AuthenticationToken` from `EXAConf`
    * The token is required as a credential for accessing the service.

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #161

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -383,22 +382,22 @@
...
-                        <configuration>
-                            <rules>
-                                <requireMavenVersion>
-                                    <version>3.3.9</version>
-                                </requireMavenVersion>
-                            </rules>
-                        </configuration>
-                    </execution>
-                </executions>
+        <execution>
```
Comment (jakobbraun) [inline pom.xml:387]: Reformat the file. Unfortunately project-keeper can't format itself

File: src/test/java/com/exasol/config/ClusterConfigAccessIT.java [hunk 2]
```diff
@@ -25,4 +25,12 @@ void testGetClusterConfiguration() {
         assertThat(this.clusterConfiguration.getBucketFsServiceConfiguration("bfsdefault").getName(),
                 equalTo("bfsdefault"));
     }
+
+    @Test
+    void testGetAuthenticationToken() {
+        this.clusterConfiguration = CONTAINER.getClusterConfiguration();
+        final String token = this.clusterConfiguration.getDefaultAuthenticationToken();
+        assertThat(token, not(emptyOrNullString()));
+        assertThat(token.length(), greaterThan(20));
```
Comment (jakobbraun) [inline src/test/java/com/exasol/config/ClusterConfigAccessIT.java:34]: That test is a bit weak. But doing a login is too much effort here?

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/config/ClusterConfigAccessIT.java (hunk 2 above)
Comment (kaklakariada) [reply src/test/java/com/exasol/config/ClusterConfigAccessIT.java:34]: I could use the bucketfs-java lib to call a method. Or I could do the call manually here. I will try.

File: doc/changes/changes_4.1.0.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Test containers for Exasol on Docker 4.1.0, released 2021-09-09
+
+Code name: RPC Support
+
+## Summary
+
+Release 4.1.0 now supports using the RPC interface of the Exasol database container.
+
+It automatically forwards the RPC port. You can also retrieve the token required for Bearer token authentication of the RPC interface.
```
Comment (redcatbear) [inline doc/changes/changes_4.1.0.md:9]:
```suggestion
It automatically forwards the RPC port. You can also retrieve the token required for bearer token authentication of the RPC interface.
```

File: doc/changes/changes_4.1.0.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+RPC allows you to manage buckets and more. You can use the version 2.2.0 and later of [bucketfs-java](https://github.com/exasol/bucketfs-java) to do this.
```
Comment (redcatbear) [inline doc/changes/changes_4.1.0.md:11]: :+1:

File: src/main/java/com/exasol/containers/ExasolContainer.java [hunk 5]
```diff
@@ -118,6 +118,7 @@ private ExasolContainer(final ExasolDockerImageReference dockerImageReference) {
         try {
             addExposedPorts(getDefaultInternalDatabasePort());
             addExposedPorts(getDefaultInternalBucketfsPort());
+            addExposedPorts(getDefaultInternalRpcPort());
```
Comment (redcatbear) [inline src/main/java/com/exasol/containers/ExasolContainer.java:121]: :+1:

**Review by redcatbear — CHANGES_REQUESTED** [review]
Editorial change requested.

File: .github/workflows/ci-build.yml [hunk 6]
```diff
@@ -52,8 +58,14 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
           SUPPORT_DIR: ${{ github.workspace }}/target/support
+      - name: Publish Test Report for ${{ matrix.docker_db_version }}
+        uses: scacap/action-surefire-report@v1
+        if: ${{ always() }}
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (morazow) [inline .github/workflows/ci-build.yml:65]: I this the one that adds Github actions annotations below for failed tests? That looks great!

**Review by morazow — CHANGES_REQUESTED** [review]
Requested minor editorial changes

File: .github/workflows/ci-build.yml (hunk 6 above)
Comment (kaklakariada) [reply .github/workflows/ci-build.yml:65]: Yes, exactly :)

**Comment by kaklakariada** [comment]
This is replaced by #166

**Pull request closed without merging**